kubernetes_state/metadata.csv

```diff
@@ -52,6 +52,7 @@ kubernetes_state.hpa.target_cpu,gauge,,,,Target CPU percentage of pods managed b
 kubernetes_state.hpa.desired_replicas,gauge,,,,Desired number of replicas of pods managed by this autoscaler,0,kubernetes,k8s_state.hpa.desired_replicas
 kubernetes_state.pod.ready,gauge,,,,"In association with the `condition` tag, whether the pod is ready to serve requests, e.g. `condition:true` keeps the pods that are in a ready state",1,kubernetes,k8s_state.pod.ready
 kubernetes_state.pod.scheduled,gauge,,,,Reports the status of the scheduling process for the pod with its tags,0,kubernetes,k8s_state.pod.scheduled
+kubernetes_state.pod.status_phase,gauge,,,,Count of current pods with their phase as a tag,0,kubernetes,k8s_state.pod.status_phase
```

Comment:
By convention, we make the value of the metric a numerical representation of the status of a pod, and then we would add a tag to the metric with the name of the pod. So maybe the value of status metric would be 2 if the tagged pod was `healthy`, 1 if it was in a `warning` state, and 0 if it was `down`.

This would allow us to see the status history of every pod when displayed on a timeseries graph.

Reply:
Sure, sounds reasonable. I'll give that a try.

Reply:
Ah, I was just thinking about this exact use case. You're right, you'll need to tag the status to each metric instance 👍 

Also, I recommend that you send this metric per pod. That way, you can get even more granularity as you'll know exactly which pod has which status. (You'll also be able to use the Host Map and the like on the web app if you need to.)

Then, on the web app, you can [count](https://docs.datadoghq.com/graphing/miscellaneous/functions/#count-not-null) the metric and filter by the status tag to get the total number of running/failed pods. As for your metric value when you're sending the metric, you can just set the value as `0` (just make a comment about how the tags represent the value and `0` is just a token value).

So in all, a rough idea of what it could look like:
```
metric_name = self.NAMESPACE + '.pod.status_phase'
tags = # your pod, namespace, and phase tags + self.custom_tags

# Metric representing the status of a pod with the tag containing the status (0 is a token value)
self.gauge(metric_name, 0, tags)
```